GraphsV2/LeafletGraph/index.js

```diff
@@ -0,0 +1,240 @@
+    initiate(props);
+
+    return () => {
+      clearTimeout(timerId);
+    }
+  }, [props.data]);
+
+  const {
+    data,
+    width,
+    properties,
+    onMarkClick,
+  } = props;
+
+  let { 
+    height,
+  } = props;
+
+  const {
+    searchBar,
+    maxZoom,
+    minZoom,
+    latitudeColumn,
+    longitudeColumn,
+    localityColumn,
+    nameColumn,
+    criticalAlarmColumn,
+    majorAlarmColumn,
+    minorAlarmColumn,
+    bootstrapStatusColumn,
+    NSGVersionColumn,
+    idColumn,
+    markerIcon,
+    filters,
+  } = properties;
+
+  const initiate = (props) => {
+    //markers = new Map();
+    setStateData(props.data);
+    setInfowindow({ data: null, position: null });
+  }
+
+  // toggle info window on marker click
+  const toggleInfoWindow = (data = null, position = null) => {
+    if (!data || !isEqual(position, infowindow.position)) {
+      setInfowindow({ data, position });
+    }
+  }
+
+  const displayNSGInfo = (data) => {
+    const displayInfo = info => {
+      return info.map((row, i) => (
+        <InfoBox key={i}>
+          <Label>{row.label}</Label>
+          <Data>&nbsp;</Data>
+          <Data>{row.text}</Data>
+        </InfoBox>
+      ))
+    }
+
+    return (
+      <InfoContainer className={'geoGraph'}>
+        {
+          displayInfo([
+            { label: 'NSG', text: data[nameColumn] },
+            { label: 'Address', text: data[localityColumn] },
+            { label: 'Bootstrap Status', text: data[bootstrapStatusColumn] || 'No status found' },
+            { label: 'NSG Version', text: data[NSGVersionColumn] || 'No version found' },
+            { label: 'Critical Alarms', text: data[criticalAlarmColumn] || 'None' },
+            { label: 'Major Alarms', text: data[majorAlarmColumn] || 'None' },
+            { label: 'Minor Alarms', text: data[minorAlarmColumn] || 'None' }
+          ])
+        }
+      </InfoContainer>
+    )
+  }
+
+  // popup info window on marker's click
+  const renderInfowindow = () => {
+    const { data } = infowindow;
+
+    return (
+      data && (
+        <Tooltip >
+          {displayNSGInfo(data)}
+        </Tooltip>
+      )
+    )
+  }
+
+  // draw markers on map
+  const renderMarkersIfNeeded = () => {
+    return stateData.filter(d => {
+      return (d[latitudeColumn] && d[longitudeColumn]);
+    }).map(d => {
+      return drawMarker({
+        data: { ...d },
+        position: [
+          d[latitudeColumn],
+          d[longitudeColumn]
+        ]
+      })
+    });
+  }
+
+  const drawMarker = ({ data, position, labelOrigin = null }) => {
+    const iconData = getIconPath(markerIcon, data);
+    const iconPerson = new L.Icon({
+      iconUrl: iconData.url,
+      iconAnchor: labelOrigin,
+    });
+
+    return (
+      <Marker
+        key={data[idColumn]}
+        position={position}
+        icon={iconPerson}
+        onClick={() => handleMarkerClick(data, onMarkClick)}
+        onMouseOver={() => toggleInfoWindow(data, position)}
```

Comment:
if we define the tooltip info in the vis config maybe we add mouse events only if those are available. We might not have to do this for all the cases.